tests/test_emulation.py

```diff
@@ -0,0 +1,238 @@
+import pytest
+from pyppeteer.device_descriptors import devices
+from pyppeteer.errors import BrowserError, PageError
+from syncer import sync
+from tests.conftest import chrome_only
+
+iPhone = devices['iPhone 6']
+iPhoneLandscape = devices['iPhone 6 landscape']
+
+
+class TestPageViewport:
+    @sync
+    async def test_gets_viewport_size(self, isolated_page):
+        """The page should get the proper viewport size"""
+        page = isolated_page
+        assert page.viewport == {"width": 800, "height": 600}
+        await page.setViewport({"width": 123, "height": 456})
+        assert page.viewport == {"width": 123, "height": 456}
+
+    @sync
+    async def test_supports_mobile_emulation(self, server, isolated_page):
+        """The page should support mobile emulation."""
+        page = isolated_page
+        await page.goto(server / 'mobile.html')
+        assert await page.evaluate("window.innerWidth") == 800
+        await page.setViewport(iPhone["viewport"])
+        assert await page.evaluate("window.innerWidth") == 375
+        await page.setViewport({"width": 400, "height": 300})
+        assert await page.evaluate("window.innerWidth") == 400
+
+    @chrome_only
+    @sync
+    async def test_supports_touch(self, server, isolated_page):
+        """Verify support of touch emulation."""
+        page = isolated_page
+        await page.goto(server / 'mobile.html')
+        assert await page.evaluate("'ontouchstart' in window") == False
+        await page.setViewport(iPhone["viewport"])
+        assert await page.evaluate("'ontouchstart' in window") == True
+        dispatchTouch = """() => {
+                let fulfill;
+                const promise = new Promise((x) => (fulfill = x));
+                window.ontouchstart = function (e) {
+                  fulfill('Received touch');
+                };
+                window.dispatchEvent(new Event('touchstart'));
+
+                fulfill('Did not receive touch');
+
+                return promise;
+            }
+        """
+        assert await page.evaluate(dispatchTouch) == 'Received touch'
+        await page.setViewport({"width": 100, "height": 100})
+        assert await page.evaluate("'ontouchstart' in window") == False
```

Comment:
I know I've writting similar, but I'd prefer one of these lines instead:
```suggestion
        assert not await page.evaluate("'ontouchstart' in window")
        assert await page.evaluate("'ontouchstart' in window") is False
```

Reply:
Fixed to `is`